content/design-patterns/hq-style-games.textile

```diff
@@ -0,0 +1,65 @@
+
+However, this timer is to be displayed by the client devices for their users (players), hence once again, this requires realtime communication between the server and all of the client devices.
+
+Further, assuming that the game can only start when at least two participants competing against each other are present, the server will need to be aware of the presence events published by clients whenever they enter or leave.
+
+Hence, let's consider three channels to solve all of the above (Of course, the actual number of channels depends on your particular use case). The illustration below gives a broad view of the data flow between players and the server for an app such as HQ trivia when implemented using Ably
+
+<a href="/images/diagrams/dp-hq-style.png" target="_blank">
+  <img src="/images/diagrams/dp-hq-style.png" style="width: 100%" alt="HQ style apps design pattern using Ably">
+</a>
+
+h2(#channels). Channels
+
+Assuming the number of clients/players for such an application is generally high, we group them into clusters based on their geographical location. Hence, we'd have three channels per cluster. For a dummy cluster 'x', let's name the three channels as below: 
+
+** "cluster­-x-­broadcast"
+** "cluster­-x-presence"
+** "cluster-x-publish"
+
+Note that the arrows in the illustration above depict the direction of flow of data in these channels.
+
+The broadcast channel can be used by the server to broadcast data simultaneously to all of the players in each cluster. This data could be the questions, correct answers to previously asked questions, timer data, etc. Basically, everything that needs to be sent to all clients alike. All the clients subscribe to the broadcast channel, but they only have subscribe permission, no publish permissions. 
+
+Next, the presence channel can be used by the clients to send their presence events to the server. In turn, the server can subscribe to this channel to be notified of this info whenever a presence event occurs. Everyone will have presence permissions on the presence channel, but not subscribe permission. This means you have the ability to query the presence on a channel or subscribe to presence events from your server, but the clients themselves won't be able to inspect/subscribe to this channel. This avoids the classic n­-squared problem i.e. 10k people entering and subscribing to 10k ‘presence-enter’ events would result in 100m messages. However, 10k people entering, generating 10k presence events only for your server results in just 20k messages. 
+
+The final one is the publish channel. This can be used by the clients in a particular cluster to publish their data to the server. This data could be their answers to each of the questions, or any other data. Everyone has publish permissions to this channel, but no subscribe rights. This publish channel can be set up to use the Reactor so you get a buffer/ shock absorber to cope with a high volume of incoming messages. 
```

Comment:
Please remember to link to our documentation, landing pages, support articles as much as possible.  The Reactor needs a link, and also needs a brief explanation of what it is.